packages/onboarding/src/setup-tailored-site-after-creation.ts

```diff
@@ -37,74 +35,61 @@ interface SetupOnboardingSiteOptions {
 	const siteDescription = select( ONBOARD_STORE ).getSelectedSiteDescription();
 	const siteLogo = select( ONBOARD_STORE ).getSelectedSiteLogo();
 
-	const postSiteSettings = ( siteId: number ) => {
-		const siteSettings = {
+	if ( siteId && flowName ) {
+		const formData: ( string | File )[][] = [];
+		const settings: {
+			blogname: string;
+			blogdescription: string;
+			launchpad_screen?: string;
+			site_intent?: string;
+		} = {
 			blogname: siteTitle,
 			blogdescription: siteDescription,
 		};
 
-		return saveSiteSettings( siteId, siteSettings );
-	};
-
-	const postSiteLogo = () => {
-		if ( ! siteLogo || ! siteId ) {
-			return Promise.resolve();
-		}
-		return uploadAndSetSiteLogo(
-			siteId,
-			new File( [ base64ImageToBlob( siteLogo ) ], 'site-logo.png' )
-		);
-	};
-
-	const setIntent = ( siteId: number, flow: string ) => {
-		if ( isNewsletterOrLinkInBioFlow( flow ) ) {
+		if ( isNewsletterOrLinkInBioFlow( flowName ) ) {
 			// link-in-bio and link-in-bio-tld are considered the same intent.
-			if ( isLinkInBioFlow( flow ) ) {
-				return setIntentOnSite( siteId.toString(), LINK_IN_BIO_FLOW );
+			if ( isLinkInBioFlow( flowName ) ) {
+				settings.site_intent = LINK_IN_BIO_FLOW;
+				if ( selectedPatternContent ) {
+					const pattern: {
+						content: string | undefined;
+						template?: string | undefined;
+					} = {
+						content: selectedPatternContent,
+					};
+					pattern.template = 'blank';
+					formData.push( [ 'pattern', JSON.stringify( pattern ) ] );
+				}
+			} else {
+				settings.site_intent = flowName;
 			}
-			return setIntentOnSite( siteId.toString(), flow );
-		}
-		return Promise.resolve();
-	};
 
-	const setLaunchpadScreen = ( siteId: number, flow: string ) => {
-		if ( isNewsletterOrLinkInBioFlow( flow ) ) {
-			return saveSiteSettings( siteId, {
-				launchpad_screen: 'full',
-			} );
+			settings.launchpad_screen = 'full';
 		}
-		return Promise.resolve();
-	};
 
-	const setPattern = async ( siteId: number, flow: string ) => {
-		if ( isLinkInBioFlow( flow ) ) {
-			await wpcomRequest( {
-				// since this is a new site, its safe to assume that homepage ID is 2
-				path: `/sites/${ siteId }/pages/2`,
-				method: 'POST',
-				apiNamespace: 'wp/v2',
-				body: { content: selectedPatternContent, template: 'blank' },
-			} );
+		formData.push( [ 'settings', JSON.stringify( settings ) ] );
 
-			return setStaticHomepageOnSite( siteId, 2 );
+		if ( siteLogo ) {
+			formData.push( [
+				'site_icon',
+				new File( [ base64ImageToBlob( siteLogo ) ], 'site-logo.png' ),
+			] );
 		}
-		return Promise.resolve();
-	};
 
-	if ( siteId && flowName ) {
-		return Promise.all( [
-			postSiteSettings( siteId ),
-			postSiteLogo(),
-			setPattern( siteId, flowName ),
-			setIntent( siteId, flowName ),
-			setLaunchpadScreen( siteId, flowName ),
-		] ).then( () => {
+		return wpcomRequest< { updated: object } >( {
+			// since this is a new site, its safe to assume that homepage ID is 2
```

Comment:
this comment is no longer relevant

Reply:
You're totally right, it survived the changes we made. Removed 👍